File: Top Interview 150/Java Code/Kadane's Algorithm/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currSum=nums[0],ans=nums[0];
        for(int i=1;i<nums.length;i++){
            if(currSum>=0) currSum+=nums[i];
            else currSum=nums[i];
            ans=Math.max(currSum,ans);
        }
        return ans;
    }
}

File: Top Interview 150/C++ Code/Kadane's Algorithm/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans=nums[0],maxsum=nums[0];
        for(int i=1;i<nums.size();i++){
            if(maxsum>=0)maxsum+=nums[i];
            else maxsum=nums[i];
            ans=max(ans,maxsum);
        }
        return ans;
    }
};
